src/puppetlabs/puppetdb/query/event_counts.clj

```diff
@@ -124,7 +124,10 @@
   "Convert an event-counts `query` and a value to `summarize_by` into a SQL string.
   A second `counts-filter` query may be provided to further reduce the results, and
   the value to `count_by` may also be specified (defaults to `resource`)."
-  ([version query [summarize_by {:keys [counts_filter count_by] :as query-options} paging-options]]
+  ([caller
```

Comment:
If you want to make that a keyword arg so it's explicit at the callsite, I'm cool with that. Just don't like functions knowing about their caller and going into a special mode as a result.